### DISCUSSION TITLE
VIM: Colors change after quitting

### DISCUSSION THREAD

**Discussion by roddolf** [JanDeDobbeleer/oh-my-posh#229] [Troubleshoot] [answered]
### Prerequisites

- [x] I have read and understand the `CONTRIBUTING` guide
- [x] I looked for duplicate issues before submitting this one

### Description

When using `vim` and then exiting, the colors of the lines before using its change of tone (see image attached).

### Environment

- Oh my Posh version: 3.51.0
- Theme: fish
- Operating System: Windows 10
- Shell: PowerShell 7.1.0
- Terminal: Windows Terminal Preview 1.5.3242.0

### Steps to Reproduce

1. [First Step] open `vim`
2. [Second Step] exit `vim`

**Expected behavior:** [What you expected to happen]
The colors remain the same.

**Actual behavior:** [What actually happened]
The colors change on the lines before using `vim`
![image](https://user-images.githubusercontent.com/14943928/101697123-12019e00-3a2c-11eb-867b-b812b3513660.png)

**Comment by JanDeDobbeleer** [comment]
@roddolf do you use the named 16 colors or hex colors?

**Comment by roddolf** [comment]
I'm using hex colors,
This is the schema set:

```json
"schemes": [
    {
        "name": "Dark",
        "black": "#000000",
        "red": "#bb0000",
        "green": "#00bb00",
        "yellow": "#bbbb00",
        "blue": "#0000bb",
        "purple": "#bb00bb",
        "cyan": "#00bbbb",
        "white": "#bbbbbb",
        "brightBlack": "#555555",
        "brightRed": "#ff5555",
        "brightGreen": "#55ff55",
        "brightYellow": "#ffff55",
        "brightBlue": "#5555ff",
        "brightPurple": "#ff55ff",
        "brightCyan": "#55ffff",
        "brightWhite": "#ffffff",
        "background": "#070707",
        "foreground": "#bbbbbb"
    }
],
```

**Comment by JanDeDobbeleer** [comment]
@roddolf I assume the theme is the default fish theme?

**Comment by roddolf** [comment]
Yes, the buildin `fish` theme.

**Comment by JanDeDobbeleer** [comment]
I'll see if I can reproduce this, chances are it's a terminal issue.

**Comment by JanDeDobbeleer** [comment]
@roddolf I tested this on the following:

- MacOS - iTerm2: no visual changes
- Windows - ConEmu: no visual changes
- Windows - Windows Terminal: **visual changes**

It's safe to say this is a Windows Terminal issue. As the colors don't change magically (they are only read within Oh my Posh, not set/changed), you have to ask for assistance there. In my tests, it even changes the previously correctly rendered colors, meaning the terminal is the one in control here.

<img width="524" alt="image" src="https://user-images.githubusercontent.com/2492783/101760536-4718f980-3adb-11eb-8984-8b361ebd58ed.png">

**Comment by JanDeDobbeleer** [comment, marked as answer]
@roddolf https://github.com/microsoft/terminal/issues/3794 and https://github.com/vim/vim/issues/5092 meaning this is indeed caused by VIM. I will move this to discussions as a way to keep track of it.
